nuxeo-services/nuxeo-platform-directory/nuxeo-platform-directory-core/src/main/java/org/nuxeo/ecm/directory/AbstractDirectory.java

```diff
@@ -112,15 +121,87 @@ public void initialize() {
         initSchemaFieldMap();
     }
 
-    protected void loadData() {
-        if (descriptor.getDataFileName() != null) {
-            try (Session session = getSession()) {
-                TransactionHelper.runInTransaction(() -> Framework.doPrivileged(() -> {
-                    Schema schema = Framework.getService(SchemaManager.class).getSchema(getSchema());
-                    DirectoryCSVLoader.loadData(descriptor.getDataFileName(),
-                            descriptor.getDataFileCharacterSeparator(), schema,
-                            ((BaseSession) session)::createEntryWithoutReferences);
-                }));
+    /**
+     * Called on Directory initialisation, depending on "createTablePolicy". Will check the Directory "duplicatePolicy"
+     * config to append or not new data.
+     *
+     * @param isTableCreated <code>false</code>if the Directory contains already some data, <code>true</code> if the
+     *            Directory (table or Collection) has been re-created (and is empty).
+     */
+    protected void loadDataOnInit(boolean isTableCreated) {
+        String dataFileName = descriptor.getDataFileName();
+        if (dataFileName != null) {
+            String dataLoadingPolicy;
+            if (isTableCreated) { // we force create since Table has just been created and empty
+                dataLoadingPolicy = ERROR_ON_DUPLICATE;
+            } else {
+                dataLoadingPolicy = descriptor.getDataLoadingPolicy();
+                if (dataLoadingPolicy.equals(NEVER_LOAD) || descriptor.isAutoincrementIdField()) {
+                    // We cannot check duplicate Id on AutoincrementIdField
+                    return;
+                }
+            }
+            Blob blob = DirectoryCSVLoader.createBlob(dataFileName);
```

Comment:
It's worth having a debug log mentioning the loading policy and the filename that's going to be used.